stingray/events.py

```diff
@@ -583,13 +580,20 @@ def to_astropy_timeseries(self):
         for attr in ['energy', 'pi']:
             if hasattr(self, attr) and getattr(self, attr) is not None:
                 data[attr] = np.asarray(getattr(self, attr))
+
         if data == {}:
             data = None
-        ts = TimeSeries(data=data, time=TimeDelta(self.time * u.s))
+
+        if self.time is not None and self.time.size > 0:
+            times = TimeDelta(self.time * u.s)
+            ts = TimeSeries(data=data, time=times)
+        else:
+            ts = TimeSeries()
         ts.meta['gti'] = self.gti
         ts.meta['mjdref'] = self.mjdref
         ts.meta['instr'] = self.instr
         ts.meta['mission'] = self.mission
+        ts.meta['header'] = self.header
         return ts
```

Comment:
I am mildly confused why this returns a `TimeSeries` object rather than an `EventList` or a `Lightcurve` object? This seems inconsistent with lots of other behaviour in stingray.

Reply:
Because this is literally the `to_astropy_timeseries()` method :)

Reply:
Oops. GitHub's folding of code that wasn't changed got me, and I thought it belonged to the `apply_deadtime` method. Sorry!